src/lxc/utils.c

```diff
@@ -2339,3 +2339,19 @@ bool lxc_nic_exists(char *nic)
 
 	return true;
 }
+
+int lxc_cloexec(int fd, bool set)
+{
+	int oflags;
+
+	oflags = fcntl(fd, F_GETFD, 0);
```

Comment:
I'm probably overthinking this but...   since liblxc is sometimes used in very highly threaded contexts - is this too racy?  (By the time we get around to our F_SETFD, someone else may have changed the base flags)

I'm probably overthinking this.

Reply:
That's actually a good point. We might simply set the flag and call it a day...

Reply:
addressed